### DISCUSSION TITLE
Mein Build (Frankenstein's Monster)

### DISCUSSION THREAD

**Discussion by DonJohnLong** [hoylabs/OpenDTU-OnBattery#905] [Show and tell]
Guten Abend Zusammen

Gerne möchte ich euch ODoB Build zeigen, ich habe dieses geniale Projekt vor ca. 2 Monaten entdeckt und war sofort angefixt und wollte auch "so was".

Als stiller Mitleser, habe ich das Projekt länger mitverfolgt, deshalb möchte mich vorgängig persönlich bei einigen Leute hier bedanken die mir besonders positiv aufgefallen sind. Ohne euch wäre das Projekt Imho. nicht dass was es heute ist.

@schlimmchen 🥇 
@spcqike
@Manos1966

und natürlich:
@helgeerbe 

Hat mega Spass gemacht das Teil zu bauen. 

Hier ein paar Bilder:

![20240421_205256](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130904315/7e614b66-6095-4fa6-a809-a31fe9910432)
![20240421_205309](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130904315/9386ef73-585c-48a1-a15b-9abf801c423c)
![20240421_205317](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130904315/863588a2-c25e-454b-a7eb-8d1419b1e6da)
![20240421_214546](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130904315/b6002326-075e-48e0-a553-912cb43bed26)

Specs:

Inverter: Hms-2000 (an separatem Stromkreis und Sicherung) 
Bms: JKBms 100A
Laderegler: Victron 150/45
Batteriezellen: EVE 100AH x16

Für Fragen und Kritik bin ich jederzeit offen.

Beste Grüsse

**Comment by Manos1966** [comment]
...und sogar mit Kuelkoerper auf der ESP32 CPU 🤩  RESPEKT!

wo hast du die Kommunikationsplatine fuer den HMS her?

**Comment by DonJohnLong** [comment]
> ...und sogar mit Kuelkoerper auf der ESP32 CPU 🤩 RESPEKT!
> 
> wo hast du die Kommunikationsplatine fuer den HMS her?

Das ist ein CMT2300A Dev Board von Ebyte auf Ali express. Ink. USB Kabel und Antenne.

https://de.aliexpress.com/item/1005005314879396.html

Ich hoffe man darf hier Links posten. Ansonsten entferne ich ihn gerne.

Grüsse

**Comment by schlimmchen** [comment]
Nice! 💪

Welche Solarmodule hast du am Start?

Was sind das für Zellen?

Wie lädst du den Inverter vor?

**Comment by DonJohnLong** [comment]
> Nice! 💪
> 
> Welche Solarmodule hast du am Start?
> 
> Was sind das für Zellen?
> 
> Wie lädst du den Inverter vor?


Die Panels sind 6x Huayao HY400-M108BSS in 3s2p Konfiguration 2.4kwp (laut Victron schon mehrfach 2.55kw geliefert)

Die Zellen sind Eve LF105 weil da die Preisleistung Imo. top ist.

Den Inverter starte ich entweder über den Victron oder wahlweise mit meinem Labornetzteil.

Beste Grüsse

![20240406_140903](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130904315/5cadc183-b231-4bb4-bc0d-69a0a88edf47)

Sind mal temporär aufgeständert, wandern dann aber auf das Vordach was noch gebaut werde muss :)

**Comment by schlimmchen** [comment]
> Die Zellen sind Eve LF105 weil da die Preisleistung Imo. top ist.

Ich hab 230er. Mein Vater bekommt jetzt 280er, für noch günstiger. Ich wünsch dir, dass du die Kapazität nicht bereust zu klein gewählt zu haben. Also ich war mir unsicher, ob 230er nicht zu groß sind, und inzwischen bin ich sehr froh, dass ich keinen kleineren Akku gewählt habe. Aber egal wie groß der Akku ist: Im Sommer ist er zu klein und im Winter zu groß. Diese Einsicht hast du vielleicht auch schon woanders gelesen.

> Den Inverter starte ich entweder über den Victron

Hm. Ist das "erlaubt"? Der Laderegler ist ausdrücklich geschützt gegen PV-Kurzschluss, von Kurzschluss an der Batterie steht da nichts. Aber es wird wohl okay sein, schließlich hat die Batterie ja ebenfalls einen äußerst geringen Innenwiderstand? Kluge Idee. Kleine Einschränkung ist dann lediglich, dass das nur tagsüber funktioniert.

> oder wahlweise mit meinem Labornetzteil.

Reichen 30V, oder hast du ein noch dickeres Labornetzteil?

> Sind mal temporär aufgeständert

Das wundert mich nicht, dass das temporär ist, der WAF (wife acceptance factor) scheint mir hier recht klein zu sein wenn die auf dem Rasen verbleiben :grin:

**Comment by greymda** [comment]
beautiful!

may i ask, what’s the blue board above the esp32 board?

**Reply by spcqike** [reply]
LM2596 step down module.

**Reply by greymda** [reply]
thanks, any insight why using it in this setup?

**Reply by spcqike** [reply]
I guess to step down 48V battery voltage to 3.3V or 5V for the esp module. So you don’t need an usb power brick

**Reply by schlimmchen** [reply]
> So you don’t need an usb power brick

In which case your ESP32 is unreachable to tell you that the BMS shut off the battery for some reason.

**Reply by DonJohnLong** [reply]
Hi, its a lm2596hvs, the hsv is important, because a normal lm 2596 will die @50+ volts.

**Reply by DonJohnLong** [reply]
@schlimmchen  Well, youre right but the bms should nerver shut down ive the system is configured right. In my case there is always around 10-15% battery charge left when the Inverter shuts down.

**Comment by greymda** [comment]
@DonJohnLong are there any fuses in your setup?

**Reply by DonJohnLong** [reply]
There are two brakers, one for the solar imput and one for the battery. No other fuses. The only cable i considered an extra fuse for was the small one that suppies the lm2596hvs with power. Otherwise everything should be covered by the main battery fuse.

**Reply by greymda** [reply]
@DonJohnLong thanks for reply. i haven’t seen in the pictures the main battery fuse, hence the question. as i’m building my own setup i am looking everywhere fir advices, so thanks

**Comment by greymda** [comment]
@DonJohnLong  can you post screenshots of the config page for DPL? thanks

**Reply by greymda** [reply]
@DonJohnLong  still applicable :)

**Reply by DonJohnLong** [reply]
Sorry didn't see your question... will do this evening.

**Reply by DonJohnLong** [reply]
> @DonJohnLong can you post screenshots of the config page for DPL? thanks

![Screenshot_20240722_201909_Chrome](https://github.com/user-attachments/assets/241acaa4-e01c-4d1e-ac9f-9fcf9f4e2157)
![Screenshot_20240722_201917_Chrome](https://github.com/user-attachments/assets/4e9db801-c127-4d16-abfb-23868a281d40)
![Screenshot_20240722_201927_Chrome](https://github.com/user-attachments/assets/49a2a4d3-2fa7-44cb-86dc-ece41639c01a)
![Screenshot_20240722_201939_Chrome](https://github.com/user-attachments/assets/c7ce5339-c13c-4b8a-a5be-e867c85f16f5)
![Screenshot_20240722_201947_Chrome](https://github.com/user-attachments/assets/cdd0ba5b-c430-4fb5-be36-1514ebe1036f)

**Reply by greymda** [reply]
thanks for that! so, interesting- you’re ignoring SoC and use Voltage? any reason why?

**Reply by DonJohnLong** [reply]
> thanks for that! so, interesting- you’re ignoring SoC and use Voltage? any reason why?

Yea, it's just more accurate in my opinion.

You sadly can't calibrate the jk bms to read high and low power draw correctly. Its always off a little bit.

![Screenshot_20240722_214226_Gallery](https://github.com/user-attachments/assets/3cea05e1-07da-492c-9b47-800079ac0d61)

**Reply by greymda** [reply]
great, thanks for getting back to me. will do the same for my setup.

**Reply by schlimmchen** [reply]
> You sadly can't calibrate the jk bms to read high and low power draw correctly.

Newer versions (firmware) have at least settings that allow to reset the SoC to 0% and 100% at specified voltages.

**Reply by greymda** [reply]
@schlimmchen mine is all over the board... it shows 0% when the voltage is ~49.5-50V

**Reply by schlimmchen** [reply]
The voltage of a single cell will trigger reset to 0% or 100%, not the total battery pack voltage. So it makes sense that the reset occurs at different battery pack voltages, depending on how far the individual cells drift.

**Reply by greymda** [reply]
@schlimmchen at 53.4V JK-BMS reports it has only 17% charge.. 

<img width="1180" alt="Screenshot 2024-07-23 at 3 43 59 PM" src="https://github.com/user-attachments/assets/e51db7bf-c313-41db-9e5e-44092c2e34e5">

**Reply by swingstate** [reply]
did you ever consider use a separate shunt?

**Reply by schlimmchen** [reply]
> at 53.4V JK-BMS reports it has only 17% charge..

And to what value did you set the capacity to in the JK BMS? Seems to me you have a much smaller battery than the JK BMS thinks you have. 17% at 53.4V is definitely way off, I agree. Mine is around 85% at this voltage.

**Reply by greymda** [reply]
@schlimmchen in the JK BMS app i've entered 20Ah, because it's a DIY battery 16s1p using 20Ah Cylindrical LiFePO4 cells.

**Reply by greymda** [reply]
> did you ever consider use a separate shunt?

well, i am thinking about now, but i'll be honest seing a shunt costing 1/2 of my battery hurts a little :(

**Comment by DonJohnLong** [comment]
Ich hab nun noch ein ändliches 2. System für meinen Vater gebaut, läuft ebenfalls top. Er hat ein perfektes kleines süd/west Dach.

Hier aber nur mit einem hm-600.

Jetzt im Sommer ist das System aber doch zimlich limitiert mit dem kleinen WR.

Falls der DPL eines Tages trotzdem zwei WR regeln kann, pack ich noch seinen 2. hm-600 dazu, der jetzt nur nutzlos herumliegt. 🥲

Hier noch ein paar Bilder:
![20240708_164239](https://github.com/user-attachments/assets/4de9c7a1-b7f3-45ea-86cf-31054e53f4ba)
![20240715_150117](https://github.com/user-attachments/assets/5f74613b-9665-4766-96d9-ae62ce53cac1)
![20240715_191005](https://github.com/user-attachments/assets/6482f782-da9d-4a3e-8fdd-2d05e1318eff)

**Reply by DerHavey1994** [reply]
Sieht echt clean aus ✌🏼
Aber kannst du bitte den Aufkleber zu "Hauptsteuerung" ändern 😬

**Reply by DonJohnLong** [reply]
> Sieht echt clean aus ✌🏼 Aber kannst du bitte den Aufkleber zu "Hauptsteuerung" ändern 😬

Haha Danke

Werd ich machen 😂
